### DISCUSSION TITLE
Reusing query results

### DISCUSSION THREAD

**Discussion by takayoshi-uchida-yoriso** [getredash/redash#6150] [Ideas]
The Athena engine version 3 allows reuse of query results.
We would like to use this in redash as well.

https://docs.aws.amazon.com/athena/latest/ug/reusing-query-results.html

**Comment by guidopetri** [comment]
You can kind of do this already by setting up 2 queries, where your first query queries the original data source and the 2nd query queries the result set of the first query [like so](https://redash.io/help/user-guide/querying/query-results-data-source). Specifically, the section titled `Cached Query Results` lets you prepend `cached_` to the 2nd query's source and it'll reuse the query results from the first query without rerunning it.

**Comment by zhongjiajie** [comment]
Thanks for @guidopetri answer, close this discussion
